includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -2554,13 +2554,14 @@ public function get_level3_data_from_order( WC_Order $order ): array {
 		$process_item  = static function( $item ) use ( $currency ) {
 			// Check to see if it is a WC_Order_Item_Product or a WC_Order_Item_Fee.
 			if ( is_a( $item, 'WC_Order_Item_Product' ) ) {
-				$subtotal   = $item->get_subtotal();
-				$product_id = $item->get_variation_id()
+				$subtotal     = $item->get_subtotal();
+				$product_id   = $item->get_variation_id()
 					? $item->get_variation_id()
 					: $item->get_product_id();
+				$product_code = substr( $product_id, 0, 12 );
```

Comment:
There are tests for `get_level3_data_from_order` [here](https://github.com/Automattic/woocommerce-payments/blob/c887217349946a5642579bb09f5ce1da73566810/tests/unit/test-class-wc-payment-gateway-wcpay.php#L441). I'd suggest to add a new test to cover this change, what do you think @nagpai ?

Reply:
Thanks for pointing me to this. It definitely makes sense for me to write a test that covers the scenario here. I will make sure to include it 👍

Reply:
Added just now @mgascam 🙌 
https://github.com/Automattic/woocommerce-payments/pull/4741/commits/4c491767347ee1992497860d249f2923177bd200